### PR TITLE
fix(nuxt): pass fully resolved path to nitro dist files

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/framework/issues/7103

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR improves the testing for inlining nuxt assets in the build:

* it improves the vite noExternal test to catch imports from `nuxt/app`.
* it passes the fully resolved path to nitro dist files (which should fix a bug with internal testing, and might possibly improve life with package managers like pnpm) - resolves https://github.com/vitejs/vite-ecosystem-ci/actions/runs/3040396486/jobs/4896394554.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/core/nitro.ts [hunk 1]
```diff
@@ -69,8 +69,7 @@ export async function initNitro (nuxt: Nuxt) {
               '@nuxt/',
               nuxt.options.buildDir
             ]),
-        'nuxt/dist',
-        'nuxt3/dist'
+        distDir
```
Comment (danielroe) [inline packages/nuxt/src/core/nitro.ts:72]:
```suggestion
        'nuxt/dist',
        'nuxt3/dist',
        distDir
```

File: packages/vite/src/server.ts [hunk 2]
```diff
@@ -72,8 +72,8 @@ export async function buildServer (ctx: ViteBuildContext) {
         /\.(es|esm|esm-browser|esm-bundler).js$/,
         '/__vue-jsx',
         '#app',
-        /(nuxt|nuxt3)\/(dist|src|app)/,
-        /@nuxt\/nitro\/(dist|src)/
+        /^nuxt(\/|$)/,
+        /(nuxt|nuxt3)\/(dist|src|app)/
```
Comment (danielroe) [inline packages/vite/src/server.ts:76]:
```suggestion
        /(^|node_modules\/)(nuxt|nuxt3)(\/(dist|src|app)|$)/
```

File: packages/vite/src/server.ts [hunk 3]
```diff
@@ -72,8 +72,7 @@ export async function buildServer (ctx: ViteBuildContext) {
         /\.(es|esm|esm-browser|esm-bundler).js$/,
         '/__vue-jsx',
         '#app',
-        /(nuxt|nuxt3)\/(dist|src|app)/,
-        /@nuxt\/nitro\/(dist|src)/
+        /(^|node_modules\/)(nuxt|nuxt3)(\/(dist|src|app)|$)/
```
Comment (danielroe) [inline packages/vite/src/server.ts:75]:
```suggestion
        /^nuxt(\/|$)/,
        /(nuxt|nuxt3)\/(dist|src|app)/
```

**Comment by danielroe** [comment]
I would like to iterate on this (moving some of this to `build.transpile` for example), but would be good to merge so https://github.com/vitejs/vite-ecosystem-ci will keep doing its good work.

**Review by pi0 — APPROVED** [review]
Thanks. LGTM (i had been pending it for rc.11 to try on edge)

File: packages/nuxt/src/core/nitro.ts [hunk 4]
```diff
@@ -70,7 +70,8 @@ export async function initNitro (nuxt: Nuxt) {
               nuxt.options.buildDir
             ]),
         'nuxt/dist',
-        'nuxt3/dist'
+        'nuxt3/dist',
+        distDir
```
Comment (pi0) [inline packages/nuxt/src/core/nitro.ts:74]: We might need to exclude dist dir for prerender (will come with another PR if needed)

**Pull request merged**